### ISSUE TITLE
Cargo running with all features by default

### ISSUE THREAD

**Issue by saghm** [intellij-rust/intellij-rust#2635]
I have my project configured to have clippy as an optional dependency behind a feature so that I can easily run it without needing to manually download it. However, clippy still only works on nightly, so the fact that the plugin use `--all-features` means that a build error is reported. It would be nice if there were a way to configure it in some way so that this doesn't happen, whether it's by being able to explicitly whitelist or blacklist the features that are used for the build or by being able to manually set the flags for the build.

**Comment by Undin** [comment]
Duplicate of #2604

**Comment by kumbayo** [comment]
Is this really a duplicate? The other issue is about `--all-targets` while this one seems to be about `--all-features` (which we only seem use when running "cargo metadata").

@saghm Can you maybe clarify the issue you are seeing?
I assume IntelliJ report an error during project opening when it runs "cargo metadata"?
This seems to be caused by #651.

**Comment by saghm** [comment]
@kumbayo Correct. You can reproduce the issue by adding the following to the Cargo.toml of a Rust project (a newly-created one with `cargo new --bin` will do):

```
[dependencies.clippy]
optional = true
version = "~0"

[features]
default = []
lint = ["clippy"]
```

When the project is opened, the `cargo metadata` command fails with the following message:

```
Cargo project update failed: Cargo execution failed (exit code 101). /usr/bin/cargo metadata --verbose --format-version 1 --all-features stdout : Updating registry `https://github.com/rust-lang/crates.io-index` Downloading utf8-ranges v1.0.1 Downloading rustc_version v0.2.3 Downloading serde v1.0.75 Downloading cc v1.0.23 Downloading regex v1.0.4 Downloading serde_derive v1.0.75 Downloading ryu v0.2.6 Downloading proc-macro2 v0.4.15 Downloading aho-corasick v0.6.8 error: unable to get packages from source Caused by: failed to parse manifest at `/home/saghm/.cargo/registry/src/github.com-1ecc6299db9ec823/clippy-0.0.212/Cargo.toml` Caused by: the cargo feature `edition` requires a nightly version of Cargo, but this is the `stable` channel stderr :
```

I should note that this isn't a huge issue for me now that clippy is provided as a rustup component, but this still should probably fixed, as using features like this is the most straightforward method for any crate to add a nightly-only feature without preventing it from being used without it on stable.
